### PR TITLE
Make HelixClusterWideAggregationTool a generic tool for helix task workflow manager

### PR DESCRIPTION
**Pull request by ankagrawal**
Make HelixClusterWideAggregationTool a generic tool for helix task workflow management of both aggregation and cloud sync tasks.

### CODE REVIEW

File: ambry-tools/src/main/java/com/github/ambry/clustermap/HelixTaskWorkflowManagerTool.java [hunk 1]
```diff
@@ -164,34 +185,20 @@ public static void main(String[] args) throws Exception {
...
+              throw new IllegalArgumentException("Invalid task type: " + config.taskType);
             }
             if (isRecurrentWorkflow) {
               workflowBuilder.setScheduleConfig(
                   ScheduleConfig.recurringFromNow(TimeUnit.MINUTES, recurrentIntervalInMinutes));
               workflowBuilder.setExpiry(TimeUnit.MINUTES.toMillis(recurrentIntervalInMinutes));
             }
             Workflow workflow = workflowBuilder.build();
-            taskDriver.start(workflow);
+            //taskDriver.start(workflow);
```
Comment (SophieGuo410) [inline ambry-tools/src/main/java/com/github/ambry/clustermap/HelixTaskWorkflowManagerTool.java:201]: Is this intended to be commented out?

File: ambry-tools/src/main/java/com/github/ambry/clustermap/HelixTaskWorkflowManagerTool.java [hunk 2]
```diff
@@ -108,6 +112,10 @@
     @Default("false")
     final boolean deleteSpecifiedWorkflow;
 
+    @Config("task.type")
+    @Default("AGGREGATION_TASK")
```
Comment (SophieGuo410) [inline ambry-tools/src/main/java/com/github/ambry/clustermap/HelixTaskWorkflowManagerTool.java:116]: Should we use AGGREGATE_TASK to align with "TaskType.AGGREGATE_TASK.name()"?

File: ambry-tools/src/main/java/com/github/ambry/clustermap/HelixTaskWorkflowManagerTool.java (hunk 1 above)
Comment (ankagrawal) [reply ambry-tools/src/main/java/com/github/ambry/clustermap/HelixTaskWorkflowManagerTool.java]: fixed. thanks for catching this.

File: ambry-tools/src/main/java/com/github/ambry/clustermap/HelixTaskWorkflowManagerTool.java (hunk 2 above)
Comment (ankagrawal) [reply ambry-tools/src/main/java/com/github/ambry/clustermap/HelixTaskWorkflowManagerTool.java]: fixed.

**Review by SophieGuo410 — APPROVED** [review]
lgtm!

File: ambry-tools/src/main/java/com/github/ambry/clustermap/HelixTaskWorkflowManagerTool.java [hunk 3]
```diff
@@ -33,21 +34,24 @@
 import org.apache.helix.task.TaskConfig;
 import org.apache.helix.task.TaskDriver;
 import org.apache.helix.task.Workflow;
+import org.bouncycastle.util.Strings;
 
 
 /**
  * This tool triggers a one-time/recurrent cluster wide stats aggregation as a generic job (executed by a random live instance).
```
Comment (jsjtzyy) [inline ambry-tools/src/main/java/com/github/ambry/clustermap/HelixTaskWorkflowManagerTool.java:41]: minor: update this java doc please

File: ambry-tools/src/main/java/com/github/ambry/clustermap/HelixTaskWorkflowManagerTool.java [hunk 4]
```diff
@@ -164,26 +185,12 @@ public static void main(String[] args) throws Exception {
...
-              jobConfigBuilder.addTaskConfigs(taskConfigs);
-              jobConfigBuilder.setCommand(aggregationCommand);
-              // add job into workflow
-              workflowBuilder.addJob(jobId, jobConfigBuilder);
+            if (config.taskType == TaskType.AGGREGATE_TASK) {
+              buildAggregationTaskWorkflow(workflowBuilder, config, isRecurrentWorkflow);
+            } else if (config.taskType == TaskType.DEPRECATED_CONTAINER_CLOUD_SYNC_TASK) {
+              buildDeprecatedContainerCloudSyncTaskWorkflow(workflowBuilder);
+            } else {
+              throw new IllegalArgumentException("Invalid task type: " + config.taskType);
```
Comment (jsjtzyy) [inline ambry-tools/src/main/java/com/github/ambry/clustermap/HelixTaskWorkflowManagerTool.java:193]: nit: consider changing this to switch-case statement for further extension in the future

**Review by jsjtzyy — APPROVED** [review]
lgtm

File: ambry-tools/src/main/java/com/github/ambry/clustermap/HelixTaskWorkflowManagerTool.java (hunk 3 above)
Comment (ankagrawal) [reply ambry-tools/src/main/java/com/github/ambry/clustermap/HelixTaskWorkflowManagerTool.java]: done.

File: ambry-tools/src/main/java/com/github/ambry/clustermap/HelixTaskWorkflowManagerTool.java (hunk 4 above)
Comment (ankagrawal) [reply ambry-tools/src/main/java/com/github/ambry/clustermap/HelixTaskWorkflowManagerTool.java]: done.

**Pull request merged**